src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -68,7 +68,21 @@
    :ruby-load-path           [schema/Str]
    :gem-path                 schema/Str
    :signeddir                schema/Str
-   :serial                   schema/Str})
+   :serial                   schema/Str
+   ;; This would be list of compile master node names (FQDNs) including MoM
+   ;; It would either be provisioned by PE or created & maintained by user (input configuration)
+   ;; OPEN: where do we get to specify defaults for the new configuration settings
```

Comment:
Probably in `config->ca-settings`

Reply:
I tried setting some defaults there but then getting tripped up in the 
"(schema/defn ^:always-validate 
  config->ca-settings :- CaSettings ....) " schema validation function. I have tried many different ways to convince it but am sure I am not doing it right. The case I was trying handle was if the new config settings are missing in the certificate-authority group in the config file then it should use the defaults. I do not want to make those config settings as optional either. Or may be if I mark them as optional in CaSettings defn that might do it - but I need to check if it would use the defaults in that case.  Am going to try that now. If that does not work I will push an update with comments around my various attempts and may be you can take a look sometime.

Reply:
ok I think I managed to make it use some defaults for the new config params for infraCRL in case they are not specified in any config files (ca.conf or puppetserver.conf). 
Had a question on the default values for some of the paths. I tried using values like "/tmp/crl/...." etc but it errors out in travis. Think I can get away with "/tmp/..." - I would expect that to exist everywhere but obviously that won't be a proper default. 
How do I specify a default that will work in either cases where it can/should use "~/.puppetlabs/etc/..." or "/etc/puppetlabs/puppet/ssl/...". 
Also question on - how you do test Clojure CA interfaces when running server from sources?